addons/binding/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/handler/Mcp23017SwitchHandler.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mcp23017.handler;
+
+import static org.openhab.binding.mcp23017.Consts.*;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.mcp23017.Consts;
+import org.openhab.binding.mcp23017.internal.GPIODataHolder;
+
+import com.pi4j.io.gpio.GpioPin;
+import com.pi4j.io.gpio.GpioPinDigitalOutput;
+import com.pi4j.io.gpio.PinState;
+
+/**
+ * The {@link Mcp23017SwitchHandler} is responsible for handling state change of digital output pin in MCP23017 chip
+ *
+ * @author Anatol Ogorek
+ */
+public class Mcp23017SwitchHandler extends AbstractMcp23017Handler {
+
+    private PinState pinState = null;
+    protected GpioPin gpioPin = null;
+
+    public Mcp23017SwitchHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received command: {} on channel {}", command.toFullString(), channelUID.getId());
+        if (channelUID.getId().equals(SWITCH_CHANNEL)) {
+            if ( Consts.ON.equals(command.toString())) {
```

Comment:
Should you really convert the command to a String before comparing, why do you do that, it is actually an enum Which can be compared safely?